src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/set-custom-user-id.mdx

```diff
@@ -0,0 +1,571 @@
+            Cordova
+        </TabsBarItem>
+        <TabsBarItem id="flutter">
+            Flutter
+        </TabsBarItem>
+        <TabsBarItem id="react">
+            React Native
+        </TabsBarItem>
+        <TabsBarItem id="xamarin">
+            Xamarin
+        </TabsBarItem>
+        <TabsBarItem id="maui">
+            .NET MAUI
+        </TabsBarItem>
+    </TabsBar>
+
+    <TabsPages>
+        <TabsPageItem id="android">
+## Syntax [#syntax]
+
+### Java [#java]
+
+```java
+NewRelic.setUserId(string $userId)
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+NewRelic.setUserId(userId: String?)
+```
+
+## Description [#description]
+
+Set a custom user identifier value to associate user sessions with analytics events and attributes.
+
+This method sets a custom user identifier value to associate all events from a mobile user's session with a specific user ID. This method can be called anytime after the New Relic Android agent starts.
+
+A user identifier is useful for several reasons. Using the [Crash analysis UI](/docs/mobile-monitoring/mobile-monitoring-ui/crashes/crash-analysis-group-filter-your-crashes), you can:
+
+* Find crashes specific to a user. Focusing on one particular crash for one user is often enough to quickly resolve the issue.
+* Filter all crashes to show the occurrences for a user (or group of users) for each crash type; inversely, filter the crash list to show those users affected by each crash type. If the user ID is represented as an email address, it's then easy to reach out and let them know when there is a solution available that addresses those crashes.
+
+If you do not want to explicitly identify users because of privacy rules, this method is still useful when tracking user segments, such as paid vs. free end users, or registered vs. unregistered users. This allows you to filter or facet on the number and types of crashes experienced by each segment, and do outreach based on that segment.
+
+For more context on how to use this API, see [Send custom attributes and events](/docs/mobile-monitoring/new-relic-mobile-android/install-configure/work-android-sdk-api#attributes-events-insights).
+
+## Parameters [#parameters]
+
+<table>
+  <thead>
+    <tr>
+      <th width={200}>
+        Parameter
+      </th>
+      <th width={200}>
+        Type
+      </th>
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$userID`
+      </td>
+
+            <td>
+        `string`
+      </td>
+
+      <td>
+        Required. Sets the user ID.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+## Return values [#return-values]
+
+Returns `true` if it succeeds, or `false` if it doesn't.
+
+## Example [#example]
+
+Here's an example to set a user identifier to associate user session with events and attributes:
+
+### Java [#java] 
+
+```java
+boolean userIdWasSet = NewRelic.setUserId("SampleUserName");
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+
+```
```

Comment:
@ndesai-newrelic Can you please provide a kotlin example equivalent to the Java example?

Reply:
val userIdWasSet = NewRelic.setUserId("SampleUserName")

Reply:
```suggestion
```kotlin
    val userIdWasSet = NewRelic.setUserId("SampleUserName")
```
```